src/algorithms/timestep/clusterexpansion.jl

```diff
@@ -0,0 +1,164 @@
+struct ClusterExpansion
+    N::Int
+end
+
+function make_time_mpo(H::MPOHamiltonian, dt::Number, alg::ClusterExpansion)
+    N = alg.N
+    lmax = N ÷ 2 # largest level
+    τ = -im * dt
+    # spaces
+    P = physicalspace(H)[1]
+    D = dim(physicalspace(H)[1]) # physical dimension
+    V = BlockTensorKit.oplus([ℂ^(D^2l) for l in 0:lmax]...)
+
+    TT = tensormaptype(ComplexSpace, 2, 2, ComplexF64)
+    O = SparseBlockTensorMap{TT}(undef, V ⊗ P ← P ⊗ V)
+
+    for n in 1:N
+        if n == 1
+            O[1, 1, 1, 1] = _make_Hamterms(H, 1, τ)
+        elseif n == 2
+            U, S, V = tsvd(_make_b(H, O, 2, τ), ((1, 2, 4), (3, 5, 6)))
+            O[1, 1, 1, 2] = permute(U * sqrt(S), ((1, 2), (3, 4)))
+            O[2, 1, 1, 1] = permute(sqrt(S) * V, ((1, 2), (3, 4)))
+        else
+            nl = n ÷ 2 # new level
+            jnl = nl + 1 # Julia indexing
+            apply_function = _make_apply_functions(O, n)
+            b = _make_b(H, O, n, τ)
+            Onew, info = lssolve(apply_function, b; verbosity=5)
+
+            (info.converged) != 1 &&
+                @warn "Did not converge when constucting the $n cluster"
+
+            if isodd(n) # linear problem
+                # assign the new level to O
+                O[jnl, 1, 1, jnl] = Onew
+            elseif iseven(n) # linear problem + svd
+                U, S, V = tsvd(Onew, (1, 2, 4), (3, 5, 6))
+                # assign the new levels to O
+                O[jnl - 1, 1, 1, jnl] = permute(U * sqrt(S), ((1, 2), (3, 4)))
```

Comment:
```suggestion
           @plansor O[jnl - 1, 1, 1, jnl][-1 -2; -3 -4] := U[-1 -2 -3; 1] * sqrt(S)[1, -4]
```
Writing this with `@plansor` calls makes it compatible with fermions, and can help with allocations in the future.